src/main/java/edu/nicholaidudakiwwarrick/advancedjava/services/DatabaseStockService.java

```diff
@@ -0,0 +1,117 @@
+import edu.nicholaidudakiwwarrick.advancedjava.util.DatabaseUtils;
+import edu.nicholaidudakiwwarrick.advancedjava.util.IntervalEnum;
+import org.joda.time.DateTime;
+
+import java.math.BigDecimal;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An implementation of the StockService interface that gets
+ * stock data from a database.
+ */
+public class DatabaseStockService implements StockService {
+
+    /**
+     * prevents instantiation - object creation delegated to ServiceFactory
+     */
+    protected DatabaseStockService() {
+
+    }
+
+    /**
+     * Returns a {@code StockQuote} instance with the specified symbol and date
+     * The symbol returned is the first quote from that day
+     *
+     * @param symbol the stock symbol of the company you want a quote for.
+     *               e.g. APPL for APPLE
+     * @param date   the date for the specific stock quote for the given symbol
+     * @throws StockServiceException if using the service generates an exception.
+     *                               If this happens, trying the service may work, depending on the actual cause of the
+     *                               error.
+     */
+    @Override
+    public final StockQuote getQuote(String symbol, DateTime date) throws StockServiceException {
+        List<StockQuote> stockQuotes = null;
+
+        // set the start and end of the passed day
+        DateTime dayStart = date.withTimeAtStartOfDay();
+        DateTime dayEnd = dayStart.plusSeconds(86399);
+
+        try {
+            Connection connection = DatabaseUtils.getConnection();
+            Statement statement = connection.createStatement();
+            String queryString = "select id from stock_symbols where symbol = '" + symbol + "'";
+            ResultSet resultSet = statement.executeQuery(queryString);
+            resultSet.next();
+            queryString = "select * from quotes where symbol_id = '" + resultSet.getInt("id") + "'"+
+                    "AND time between '" + dayStart.toString(StockQuote.DATE_PATTERN) + "' and '" +
+                    dayEnd.toString(StockQuote.DATE_PATTERN) + "'";
+            resultSet = statement.executeQuery(queryString);
+            stockQuotes = new ArrayList<>(resultSet.getFetchSize());
+            while(resultSet.next()) {
+                Timestamp timeStamp = resultSet.getTimestamp("time");
+                DateTime time = new DateTime(timeStamp);
+                BigDecimal price = resultSet.getBigDecimal("price");
+                stockQuotes.add(new StockQuote(symbol, price, time));
+            }
+        } catch (DatabaseConnectionException | SQLException exception) {
+            throw new StockServiceException(exception.getMessage(), exception);
+        }
+        if (stockQuotes.isEmpty()) {
+            throw new StockServiceException("There is no stock data for:" + symbol);
+        }
+        return stockQuotes.get(0);
```

Comment:
this is not ideal since the list could be empty. 
also,  not ideal because why pick the first stock to return?